tests/src/test/java/org/eclipse/hono/tests/registry/DeviceManagementIT.java

```diff
@@ -423,6 +430,209 @@ public void testDeregisterDeviceFailsForNonExistingDevice(final VertxTestContext
+            final String deviceId2 = helper.getRandomDeviceId(tenantId);
+            final Device device1 = new Device().setExtensions(Map.of("id", "aaa"));
+            final Device device2 = new Device().setExtensions(Map.of("id", "bbb"));
+            final int pageSize = 1;
+            final int pageOffset = 1;
+            final String sortJson = getSortJson("/ext/id", "desc");
+
+            CompositeFuture.all(registry.registerDevice(tenantId, deviceId1, device1),
+                    registry.registerDevice(tenantId, deviceId2, device2))
+                    .compose(ok -> registry.searchDevices(tenantId, Optional.of(pageSize), Optional.of(pageOffset),
+                            null, List.of(sortJson), HttpURLConnection.HTTP_OK))
+                    .onComplete(ctx.succeeding(httpResponse -> {
+                        ctx.verify(() -> {
+                            final JsonArray response = httpResponse.bodyAsJsonArray();
+                            assertThat(response.size()).isEqualTo(1);
+
+                            final JsonObject deviceObject = response.getJsonObject(0);
+                            assertThat(deviceObject.getString("id")).isEqualTo(deviceId1);
+                        });
+                        ctx.completeNow();
+                    }));
+        }
+
+        /**
+         * Verifies that a request to search devices fails when filterJson is invalid.
+         *
+         * @param ctx The vert.x test context.
+         */
+        @Test
+        public void testSearchDevicesFailsForInvalidFilterJson(final VertxTestContext ctx) {
+
+            registry.registerDevice(tenantId, deviceId)
+                    .compose(ok -> registry.searchDevices(tenantId, Optional.empty(), Optional.empty(),
+                            List.of("Invalid filterJson"), null, HttpURLConnection.HTTP_BAD_REQUEST))
+                    .onComplete(ctx.completing());
+        }
+
+        /**
+         * Verifies that a request to search devices with multiple filters succeeds and matching devices are found.
+         *
+         * @param ctx The vert.x test context.
+         */
+        @Test
+        public void testSearchDevicesWithMutlipleFilters(final VertxTestContext ctx) {
+            final Device device1 = new Device().setEnabled(false).setExtensions(Map.of("id", "1"));
+            final Device device2 = new Device().setEnabled(true).setExtensions(Map.of("id", "2"));
+            final String filterJson1 = getFilterJson("/ext/id", "1", "eq");
+            final String filterJson2 = getFilterJson("/enabled", true, "eq");
+
+            CompositeFuture.all(registry.registerDevice(tenantId, device1), registry.registerDevice(tenantId, device2))
+                    .compose(ok -> registry.searchDevices(tenantId, Optional.empty(), Optional.empty(),
+                            List.of(filterJson1, filterJson2), null, HttpURLConnection.HTTP_NOT_FOUND))
+                    .onComplete(ctx.completing());
```

Comment:
how about adding a step verifying that a combination of other filters actually yields a result?